diff --git a/.cargo/config.toml b/.cargo/config.toml
--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -3,6 +3,7 @@
 [target.x86_64-unknown-none]
 rustflags = [
+	"--cfg", 'target_min_page="4kib"',
 	"-C", "force-frame-pointers",
 	"--cfg", "aes_force_soft",
 	"--cfg", 'aes_backend="soft"',
@@ -11,6 +12,7 @@ rustflags = [

 [target.x86_64-unknown-linux-gnu]
 rustflags = [
+       "--cfg", 'target_min_page="4kib"',
        "-C", "code-model=kernel",
 ]
diff --git a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -53,7 +53,7 @@ bootdefs = { path = "boot/defs" }
 bootimg = { path = "boot/bootimg" }
 cpuarch = { path = "cpuarch" }
-paging = { path = "paging" }
+paging = { package = "svsm-paging", path = "paging" }
 test = { path = "test" }
 svsm = { path = "kernel" }
 elf = { path = "elf" }
@@ -129,10 +129,10 @@ zerocopy = { version = "0.8.2", features = ["derive"] }
 zeroize = { version = "1.8.2", default-features = false }

 # Verus repos
-verus_builtin = { version = "=0.0.0-2026-05-17-0151", default-features = false }
-verus_builtin_macros = { version = "=0.0.0-2026-07-12-0122", features = ["vpanic"], default-features = false }
-verus_state_machines_macros = { version = "=0.0.0-2026-06-14-0213", default-features = false }
-vstd = { version = "=0.0.0-2026-07-12-0122", features = ["alloc", "allow_panic"], default-features = false }
+verus_builtin = { version = "=0.0.0-2026-08-02-0125", default-features = false }
+verus_builtin_macros = { version = "=0.0.0-2026-08-02-0125", features = ["vpanic"], default-features = false }
+verus_state_machines_macros = { version = "=0.0.0-2026-08-02-0125", default-features = false }
+vstd = { version = "=0.0.0-2026-08-02-0125", features = ["alloc", "allow_panic"], default-features = false }

 # Verification libs
 verify_proof = { path = "verification/verify_proof", default-features = false  }
diff --git a/paging/Cargo.toml b/paging/Cargo.toml
--- a/paging/Cargo.toml
+++ b/paging/Cargo.toml
@@ -1,9 +1,12 @@
 [package]
-name = "paging"
+name = "svsm-paging"
 version = "0.1.0"
 edition.workspace = true
 description = "Generic page table structures for Rust OS projects"

+[lib]
+name = "paging"
+
 [dependencies]
 bitflags.workspace = true
 zerocopy.workspace = true
diff --git a/kernel/Cargo.toml b/kernel/Cargo.toml
--- a/kernel/Cargo.toml
+++ b/kernel/Cargo.toml
@@ -21,6 +21,7 @@ bootimg.workspace = true
 cpuarch.workspace = true
 paging.workspace = true
+verismo_paging = { package = "paging", path = "__VERISMO_PAGING__", default-features = false, features = ["use_ad"] }

 libaproxy = { workspace = true, optional = true }
 elf.workspace = true
diff --git a/kernel/src/mm/mod.rs b/kernel/src/mm/mod.rs
--- a/kernel/src/mm/mod.rs
+++ b/kernel/src/mm/mod.rs
@@ -23,6 +23,7 @@ pub mod ptguards;
 pub mod ro_after_init;
 pub mod validate;
+pub mod verismo_paging;
 pub mod virtualrange;
 pub mod vm;

diff --git a/kernel/src/mm/alloc.rs b/kernel/src/mm/alloc.rs
--- a/kernel/src/mm/alloc.rs
+++ b/kernel/src/mm/alloc.rs
@@ -2402,6 +2402,11 @@ static ALLOCATOR: SvsmAllocator = SvsmAllocator::new();

+pub(crate) fn root_memory_mapping() -> (PhysAddr, VirtAddr, usize) {
+    let root = ROOT_MEM.lock();
+    (root.start_phys, root.start_virt, root.page_count * PAGE_SIZE)
+}
+
 /// Initializes the root memory region with the specified physical start
 /// address, virtual start address, and page count.
 pub fn root_mem_init(pstart: PhysAddr, vstart: VirtAddr, page_count: usize, allocated: usize) {
